src/olympia/devhub/tasks.py

```diff
@@ -88,7 +89,9 @@ def create_version_for_upload(addon, upload):
 
         log.info('Creating version for {upload_uuid} that passed '
                  'validation'.format(upload_uuid=upload.uuid))
-        version = Version.from_upload(upload, addon, [amo.PLATFORM_ALL.id])
+        beta = is_beta(upload.version)
```

Comment:
Maybe it's too paranoid but since you're relying on the magical `upload.version`, it would be safer to raise an exception here if that attribute is None.